packages/neon/neon/lib/src/blocs/timer.dart

```diff
@@ -0,0 +1,72 @@
+part of '../../neon.dart';
+
+abstract class TimerBlocEvents {
+  /// Register a [callback] that will be called periodically.
+  /// The time between the executions is defined by the [duration].
+  TimerBlocTimer registerTimer(final Duration duration, final VoidCallback callback);
+
+  /// Unregister a timer that has been previously registered with the bloc.
+  /// You can also use [TimerBlocTimer.cancel].
+  void unregisterTimer(final TimerBlocTimer timer);
+}
+
+abstract class TimerBlocStates {}
+
+/// Execute callbacks at defined periodic intervals.
+/// Components can register their callbacks and everything with the same periodicity will be executed at the same time.
+///
+/// The [TimerBloc] is a singleton.
+/// Sub-second timers are not supported.
+class TimerBloc extends Bloc implements TimerBlocEvents, TimerBlocStates {
+  factory TimerBloc() => _singleton;
+  TimerBloc._();
+
+  static final TimerBloc _singleton = TimerBloc._();
+
+  final Map<int, Timer> _timers = {};
+  final Map<int, List<VoidCallback>> _callbacks = {};
+
+  @override
+  void dispose() {
+    for (final timer in _timers.values) {
+      timer.cancel();
+    }
+  }
+
+  @override
+  TimerBlocTimer registerTimer(final Duration duration, final VoidCallback callback) {
+    if (_timers[duration.inSeconds] == null) {
+      _timers[duration.inSeconds] = Timer.periodic(duration, (final _) {
+        for (final callback in _callbacks[duration.inSeconds]!) {
+          callback();
+        }
+      });
+      _callbacks[duration.inSeconds] = [callback];
+    } else {
+      _callbacks[duration.inSeconds]!.add(callback);
+    }
+    return TimerBlocTimer(duration, callback);
+  }
+
+  @override
+  void unregisterTimer(final TimerBlocTimer timer) {
+    if (_timers[timer.duration.inSeconds] != null) {
+      _callbacks[timer.duration.inSeconds] =
+          _callbacks[timer.duration.inSeconds]!.where((final cb) => cb != timer.callback).toList();
+    }
```

Comment:
wouldn't `callbacks[timer.duration.inSeconds]?.remove(timer.callback);` also do the trick?